### PR TITLE
Ad img refactor

### PR DESCRIPTION
**Pull request by curioussavage**
:eyeglasses: @ajacksified

### CODE REVIEW

File: src/views/components/ListingContent.jsx [hunk 1]
```diff
@@ -48,8 +49,8 @@ function _aspectRatioClass(ratio) {
   if (!ratio) {
     return  'aspect-ratio-' + 16 + 'x' + 9;
   }
-  var w = MyMath.round(ratio * _HEIGHT, _INCREMENT);
-  var euclid = MyMath.euclid(w, _HEIGHT);
+  let w = MyMath.round(ratio * _HEIGHT, _INCREMENT);
```
Comment (ajacksified) [inline src/views/components/ListingContent.jsx]: prefer `const` to `let`

File: src/views/components/ListingContent.jsx [hunk 2]
```diff
@@ -61,17 +62,39 @@ function _wrapSelftextExpand(fn) {
   }
 }
 
+function _isPlayable(listing) {
+  let media = listing.media
```
Comment (ajacksified) [inline src/views/components/ListingContent.jsx]: you need semicolons here and in many places below

File: src/views/components/ListingContent.jsx [hunk 3]
```diff
@@ -61,17 +62,39 @@ function _wrapSelftextExpand(fn) {
...
+  let urlObj = URL.parse(url);
+  urlObj.protocol = https ? 'https:' : urlObj.protocol;
+  return URL.format(urlObj);
+}
+
 class ListingContent extends BaseComponent {
   constructor(props) {
     super(props);
+
+    // we want things to autoplay in compact mode
```
Comment (ajacksified) [inline src/views/components/ListingContent.jsx]: in _expanded_ compact mode

File: src/views/components/ListingContent.jsx [hunk 4]
```diff
@@ -193,50 +180,45 @@ class ListingContent extends BaseComponent {
...
-        style.backgroundImage = 'url(' + a.href + ')';
-      }
+    let compact = this._isCompact();
+    let isNSFW = ListingContent.isNSFW(props.listing);
+
+    let style = {};
+    let loaded = props.loaded;
+
+    if (playable && !(isNSFW && !props.showNSFW)) {
+      let playIconNode = <span className='icon-play-circled'>{' '}</span>;
```
Comment (ajacksified) [inline src/views/components/ListingContent.jsx]: you need to define your `let` higher up, or you get an `undefined playIconNode` exception.

**Comment by ajacksified** [comment]
:nail_care:

File: src/views/components/ListingContent.jsx (hunk 4 above)
Comment (curioussavage) [inline src/views/components/ListingContent.jsx]: good catch, I think I noticed that it wasn't defined earlier but it still works for me which is wierd.

**Comment by curioussavage** [comment]
:haircut:

**Comment by ajacksified** [comment]
:fish:

**Pull request merged**